document/AnnexE-Determinism

```diff
@@ -0,0 +1,199 @@
+file paths to lowercase or to uppercase, using a locale-insensitive case conversion:
+
+```
+"uri": "file:///C:/code/file1.c"
+```
+
+~IsoNote
+
+On the .NET platform, the recommendation is to convert strings to uppercase
+using the method `String.ToUpperInvariant`, which would result in:
+
+```
+"uri": "file:///C:/CODE/FILE1.C"
+```
+~
+
+## Compensating for non-deterministic output { - #Annex-E-Compensating }
+
+If an analysis tool does not produce deterministic output,
+a build system can add additional processing steps to compensate.
+
+There are two scenarios to consider:
+
+1. Log equality is determined by a simple comparison of file contents, or by comparing file hashes.
+2. Log equality is determined by an "intelligent" comparison.
+
+In the first scenario, a post-processing step could
+produce deterministic output by creating a new file that omits non-deterministic elements,
+reorders array elements and object properties,
+removes file path prefixes and introduces `uriBaseId` properties,
+and adjusts the casing of case-insensitive string values.
+
+In the second scenario, a post-processing step could intelligently compare the newly produced log
+to the log from a previous build by ignoring non-deterministic elements,
+ensuring that arrays have the same elements regardless of order,
+ignoring file path prefixes,
+and performing a case-insensitive comparison.
+
+## Interaction between determinism and baselining { - #Annex-E-Baselining }
+
+SARIF's baselining feature poses a particular challenge for determinism.
+We illustrate the problem with the following scenario:
+
+On a particular date, a project's nightly build runs an analysis tool ToolX,
+which produces a log file, say, `log_20160614.sarif`.
+The next day, a developer modifies one of the files scanned by the tool
+in a way that introduces a new problem.
+That night, the nightly build tool runs again, this time producing a log file which
+compares the current set of results to those that appeared in the previous run:
+```
+ToolX --input a.c b.c --baseline log_20160614.sarif --output log_20160615.sarif
+```
+Because a new problem has been introduced, `log_20160614.sarif` will contain a `result` object
+whose `baselineState` is `"new"`.
+The next night, without any further changes to the source files, the tool is run yet again:
+```
+ToolX --input a.c b.c --baseline log_20160615.sarif --output log_20160616.sarif
+```
+The `result` object that first appeared in `log_20160615.sarif` still appears in `log_20160616.sarif`,
+but since it existed in the baseline, its `baselineState` will now be `"existing"`.
+
+The result is that even though none of the analysis target files have changed,
+the log file has changed.
+
+Strictly speaking, this does not violate determinism. After all, the baseline file has changed,
+and the baseline file is one of the inputs to the analysis.
+But from a practical standpoint, this is still a problem, because... why?
```

Comment:
Or maybe I'm missing your point. Are you simply saying that we should _warn_ people that this will occasionally happen, if they both (a) use the baselining feature, and (b) use a build system that makes decisions base on a non-intelligent comparison? I would agree with that.